db/db.go

```diff
@@ -245,9 +214,142 @@ func (db *DB) execImpl(basis types.Ref, function string, args types.List) (newDa
 	return newData, newDataChecksum, output, isWrite, nil
 }
 
+func (db *DB) NewTransaction() *Transaction {
+	return &Transaction{
+		db:   db,
+		noms: db.noms,
+		head: db.head,
+		me:   db.head.Data(db.noms).Edit(),
+		Args: types.NewList(db.noms),
+	}
+}
+
 func (db *DB) lock() func() {
 	db.mu.Lock()
 	return func() {
 		db.mu.Unlock()
 	}
 }
+
+type Transaction struct {
+	db       *DB
+	noms     datas.Database
+	head     Commit
+	me       *kv.MapEditor
+	didWrite bool
+	closed   bool
+	Name     string
```

Comment:
For now I wanted it mutable but at one point I had an overloaded `NewTransactionWithArgs` function. I will make it that way for now because it is more restrictive.